src/components/tic-tac-toe.tsx

```diff
@@ -0,0 +1,377 @@
+          className={css(`
+            margin-left: 20px;
+        `)}
+        >
+          <div>{status}</div>
+          <ol
+            className={css(`
+              padding-left: 30px;
+
+          `)}
+          >
+            {moves}
+          </ol>
+          <Toggle
+            isToggleOn={this.state.isToggleOn}
+            onClick={() => this.toggleHandler()}
+          />
+        </div>
+      </div>
+    );
+  }
+}
+
+function calculateWinner(squares: string[]) {
+  const lines = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+
+  for (const [a, b, c] of lines) {
+    if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
+      return [a, b, c];
+    }
+  }
+
+  return null;
+}
+
+// Rteurns an array of the indicies of all occurances
+function getAllIndexOF(arr: any[], val: any): number[] {
+  const indicies = [];
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] === val) {
+      indicies.push(i);
+    }
+  }
+  return indicies;
+}
+
+/**
+ * Returns a random integer between min (inclusive) and max (inclusive)
+ * Using Math.round() will give you a non-uniform distribution!
+ */
+function getRandomInt(min: number, max: number): number {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function chooseRandom(arr: any[]): any {
+  return arr[getRandomInt(0, arr.length - 1)];
+}
+
+function notFalsy(value: any): boolean {
+  return !!value;
+}
+
+function notNull(value: any): boolean {
+  return value !== null;
+}
+
+// Player is X
+// Computer is O
+function calculateNextMove(
+  squares: string[],
+  xIsNext: boolean,
+): { maxOrMin: number; index: number } {
+  // First check for winning states in recursion
+  if (calculateWinner(squares)) {
+    const max = squares[calculateWinner(squares)[0]] === 'X' ? -1 : 1;
+    return { maxOrMin: max, index: null };
+  }
+
+  if (squares.slice().filter(notFalsy).length === 9) {
+    return { maxOrMin: 0, index: null };
+  }
+
+  // If I have to make a move
+  const weights = Array(squares.length).fill(null);
+  for (let i = 0; i < squares.length; i++) {
+    if (squares[i]) {
+      continue;
+    }
+    const newSquares = squares.slice();
+    newSquares[i] = xIsNext ? 'X' : 'O';
+    weights[i] = calculateNextMove(newSquares, !xIsNext).maxOrMin;
+  }
+
+  const maxOrMin = xIsNext
+    ? Math.min(...weights.slice().filter(notNull))
+    : Math.max(...weights.slice().filter(notNull));
+  const possibleMax = getAllIndexOF(weights, maxOrMin);
+  const chosenOne = chooseRandom(possibleMax);
+  // if (squares.slice().filter(notFalsy).length === 1) {
+  //   console.log(`squares: ${squares}`);
+  //   console.log(`Weights: ${weights}`);
+  //   console.log(`PossibleMAx: ${possibleMax}`);
+  // }
+  return { maxOrMin, index: chosenOne };
+}
+
+// ========================================
+module.exports = { Game, Square, Board, Row, calculateNextMove };

```

Comment:
This is an *extremely* long file, with many different things in it. Please extract some of it into standalone modules.